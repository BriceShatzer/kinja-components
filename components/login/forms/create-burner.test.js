import * as React from 'react';
import { shallow } from 'enzyme';
import CreateBurner from './create-burner';

window.grecaptcha = {
	render: jest.fn(),
	reset: jest.fn()
};

describe('Create Burner form', () => {
	it('should render by default and render captcha', () => {
		const wrapper = shallow(<CreateBurner translate={str => str}/>).dive();
		expect(wrapper).toMatchSnapshot();
		expect(window.grecaptcha.render).toHaveBeenCalled();
	});
	it('should display username taken error and reset captcha', () => {
		const wrapper = shallow(<CreateBurner translate={str => str} />).dive();
		wrapper.setProps({
			userNameTaken: true
		});
		expect(wrapper).toMatchSnapshot();
		expect(window.grecaptcha.reset).toHaveBeenCalled();
	});
	it('should display error when given an invalid screen name', () => {
		const handler = jest.fn();
		const wrapper = shallow(<CreateBurner onCreateClick={handler} onInputChange={jest.fn()} translate={str => str} />).dive();
		wrapper.find('ScreenNameInput').simulate('change', null);
		expect(wrapper.find('Button').prop('disabled')).toBe(true);
		expect(wrapper.state('error')).toBe(true);
	});
});