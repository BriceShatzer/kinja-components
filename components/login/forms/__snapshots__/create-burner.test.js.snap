// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Burner form should display username taken error and reset captcha 1`] = `
<EnsureDefaultTheme>
  <common__Form
    onSubmit={[Function]}
  >
    <common__MobileBackButton>
      <a
        role="button"
        tabIndex="0"
      >
        <ArrowLeftIcon />
         
        Back
      </a>
    </common__MobileBackButton>
    <common__MobileH1>
      Create a burner account
    </common__MobileH1>
    <common__CloseButton>
      <a
        role="button"
        tabIndex="0"
      >
        <CloseIcon />
      </a>
    </common__CloseButton>
    <div>
      <ScreenNameInput
        inlineHelp="You can change this later in Account Settings"
        label="Set your User Name"
        onChange={[Function]}
        placeholder="Set your User Name here"
        translate={[Function]}
        userNameTaken={true}
      />
      <create-burner__Captcha
        className="g-recaptcha"
        id="g-recaptcha"
      />
    </div>
    <common__Footer>
      <Button
        disabled={true}
        icon={<ArrowRightIcon />}
        label="Create Account"
        labelPosition="before"
        onClick={[Function]}
        weight="primary"
      />
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "
			By creating an account, you’re agreeing to our
			 <a href=\\"https://g-omedia.com/terms-of-service/\\">
				 Terms of Use
			</a> and
			<a href=\\"https://g-omedia.com/privacy-policy/\\">
				Privacy Policy
			</a>.",
          }
        }
      />
    </common__Footer>
  </common__Form>
</EnsureDefaultTheme>
`;

exports[`Create Burner form should render by default and render captcha 1`] = `
<EnsureDefaultTheme>
  <common__Form
    onSubmit={[Function]}
  >
    <common__MobileBackButton>
      <a
        role="button"
        tabIndex="0"
      >
        <ArrowLeftIcon />
         
        Back
      </a>
    </common__MobileBackButton>
    <common__MobileH1>
      Create a burner account
    </common__MobileH1>
    <common__CloseButton>
      <a
        role="button"
        tabIndex="0"
      >
        <CloseIcon />
      </a>
    </common__CloseButton>
    <div>
      <ScreenNameInput
        inlineHelp="You can change this later in Account Settings"
        label="Set your User Name"
        onChange={[Function]}
        placeholder="Set your User Name here"
        translate={[Function]}
        userNameTaken={false}
      />
      <create-burner__Captcha
        className="g-recaptcha"
        id="g-recaptcha"
      />
    </div>
    <common__Footer>
      <Button
        disabled={true}
        icon={<ArrowRightIcon />}
        label="Create Account"
        labelPosition="before"
        onClick={[Function]}
        weight="primary"
      />
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "
			By creating an account, you’re agreeing to our
			 <a href=\\"https://g-omedia.com/terms-of-service/\\">
				 Terms of Use
			</a> and
			<a href=\\"https://g-omedia.com/privacy-policy/\\">
				Privacy Policy
			</a>.",
          }
        }
      />
    </common__Footer>
  </common__Form>
</EnsureDefaultTheme>
`;
